scrunch/datasets.py

```diff
@@ -680,7 +683,42 @@ def __readonly__(self, *args, **kwargs):
     del __readonly__
 
 
-class Dataset(object):
+class DatasetVariablesMixin(collections.Mapping):
```

Comment:
I am not sure about this one. __getitem__ has logic that shouldn't be on a iterator in this case